src/com/google/cloud/datastore/mlang.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Refine Property and DOT to produce datastore property.
+//
+
+foam.CLASS({
+  refines: 'foam.core.Property',
+
+  methods: [
+    {
+      name: 'toOwnDatastoreProperty',
+      documentation:
+      function() {/*
+                     Provides a PropertyType.name for the Cloud Datastore
+                     REST API. I.e., "name" key in
+                     https://cloud.google.com/datastore/docs/reference/rest/v1/projects/runQuery#PropertyReference
+                  */},
+      code: function() { return this.name; }
+    },
+    {
+      name: 'toDatastoreProperty',
```

Comment:
Those that are compatible with expressions can be methods, but the 'datastoreType' could just be a Property.